src/nvim/api/vim.c

```diff
@@ -35,13 +35,18 @@
 #include "nvim/os/input.h"
 #include "nvim/viml/parser/expressions.h"
 #include "nvim/viml/parser/parser.h"
+#include "nvim/ui.h"
 
 #define LINE_BUFFER_SIZE 4096
+#define MAX_UI_COUNT 16
 
 #ifdef INCLUDE_GENERATED_DECLARATIONS
 # include "api/vim.c.generated.h"
 #endif
 
+extern UI *uis[MAX_UI_COUNT];
+extern size_t ui_count;
```

Comment:
This is what I wanted to avoid. I would suggest instead creating a wrapper function in ui.h, and just call it from `nvim_list_uis()` (which lives in vim.c):

```c
Array nvim_list_uis(void)
  FUNC_API_SINCE(4)
{
  return ui_list();
}

Reply:
Okay, I get it now. will implement it today. thanks!